test/pipeline_test_runner.cc

```diff
@@ -421,9 +426,6 @@ TEST_CASE("PerPhaseTest") { // NOLINT
         ErrorAssertion::checkAll(test.sourceFileContents, RangeAssertion::getErrorAssertions(assertions), diagnostics);
     }
 
-    // Allow later phases to have errors that we didn't test for
-    errorQueue->drainAllErrors();
```

Comment:
I think this still needs to `flushAllErrors` on the queue and `drainErrors` on the collector.